src/main/java/org/eclipse/yasson/internal/deserializer/DeserializationModelCreator.java

```diff
@@ -233,7 +233,7 @@ private ModelDeserializer<JsonParser> createObjectDeserializer(LinkedList<Type>
             if (creatorModel.getCustomization().isRequired()) {
                 defaultCreatorValues.put(parameterName, new RequiredCreatorParameter(parameterName));
             } else {
-                Class<?> rawParamType = ReflectionUtils.getRawType(creatorModel.getType());
+                Class<?> rawParamType = ReflectionUtils.getOptionalRawType(creatorModel.getType()).orElse(Object.class);
```

Comment:
I'm not 100% sure if this is the correct place to fix this. I did attempt to fix it by defaulting to `Optional.of(Object.class)` in the `ReflecionUtils.getOptionalRawType()`. However, that did break some tests and I wasn't sure if that was correct either.